### ISSUE TITLE
Not opening on Windows 11

### ISSUE THREAD

**Issue by Soitora** [flozz/yoga-image-optimizer#43]
### Bug description
Using Portable and MSI installer, all version I've tried (down to, and including, 1.1.2) does not open at all.

#### What happened

Simply use Portable or MSI installer

#### What was expected

Program to open the GUI

#### Step to reproduce

* [X] The bug is always reproducible
* [ ] The bug seems to appear randomly

Steps to reproduce:

- Install MSi or Extract Portable files
- Attempt to open the program through shortcut or .exe

### System information

* Operating system (with version): [Ubuntu 20.04, ArchLinux, Windows 10,...]
* How did you installed the software?
    * [ ] From PyPI (`pip3 install yoga-image-optimizer`)
    * [ ] From source
    * [X] Windows Installer
    * [X] Windows Zip
    * [ ] Other (please provide any useful information like command you use, a link to a documentation,...)

* YOGA Image Optimizer version: 1.2.2 -> 1.1.2

### Extra information

I did not find logs anywhere, it could most likely help but I do not know where to search.

**Comment by flozz** [comment]
Hello,

Thank you for your report. I tried YOGA Image Optimizer v1.2.2 on a Windows 11 VM and it worked for me. So I am not able to reproduce and to debug your issue by myself.

Sadly there is no easy way to get application output on Windows . If the program is compiled with no console support, there will be no log, and if it is compiled with console it will always open a terminal... I may try to implement my own logging one day but not now...

Anyway, I am building for you a version that opens the console so you will be able to have log to report here :)

**Comment by Soitora** [comment]
> Hello,
> 
> Thank you for your report. I tried YOGA Image Optimizer v1.2.2 on a Windows 11 VM and it worked for me. So I am not able to reproduce and to debug your issue by myself.

That's very odd, I might add that I'm on Developer preview version of Windows 11

Other than that I'm unsure of what could affect it

> Anyway, I am building for you a version that opens the console so you will be able to have log to report here :)

Thank you kindly 🙏

**Comment by flozz** [comment]
The build is here:

* https://nextcloud.flozz.org/s/nAZ6cSgnm5Q3Pfq

You will have to

* Decompress the ZIP archive
* Open a terminal in the folder that contains the software
* Run it from the CLI (so if the app crashes you will still have the logs)

![Capture d’écran du 2023-08-27 11-39-19](https://github.com/flozz/yoga-image-optimizer/assets/971438/40d4d469-d7c2-490c-b944-f50b3907c95a)

**Comment by Soitora** [comment]
Thanks

I extracted it to `C:\Users\Soitora\Downloads\yoga-image-optimizer_1.2.2_win64` and called `.\yoga-image-optimizer.exe` using `Windows Terminal` and `Powershell Core 7`

Here is my output:
```bash
❯ .\yoga-image-optimizer.exe
Traceback (most recent call last):
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\yoga-image-optimizer.py", line 16, in <module>
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\yoga_image_optimizer\__main__.py", line 4, in <module yoga_image_optimizer.__main__>
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\yoga_image_optimizer\application.py", line 11, in <module yoga_image_optimizer.application>
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\yoga_image_optimizer\helpers.py", line 12, in <module yoga_image_optimizer.helpers>
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\yoga_image_optimizer\translation.py", line 11, in <module yoga_image_optimizer.translation>
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\os.py", line 685, in __setitem__
  File "C:\Users\Soitora\DOWNLO~1\YOGA-I~1.2_W\os.py", line 743, in check_str
TypeError: str expected, not NoneType
```

**Comment by flozz** [comment]
Ok, it seems to be a very simple bug at least :)

It seems that here, `getLocale()` returns `None` instead of a valid language, and then it breaks when I try to insert the value in the environment:

```python
    language, encoding = locale.getlocale()
    os.environ["LANG"] = language
```

I do not know that `getLocale()` could return a non-string value. I will add a check to prevent the app from crashing here.

Can you tell me what is the locale (the language) configured on your Windows?

**Comment by flozz** [comment]
I rebuilt the code with a fix. Can you try it works? :)

* https://nextcloud.flozz.org/s/XoiKtx4oLWKgWMp  [the link expires by end 2023]

**Comment by Soitora** [comment]
Awesome, seems to work!

Language settings you asked for:

![image](https://github.com/flozz/yoga-image-optimizer/assets/10836780/eff932da-6bc0-43f9-8063-6e52061e09dd)

How it looks:

![image](https://github.com/flozz/yoga-image-optimizer/assets/10836780/22385f83-1966-4e9b-a98c-b944289ce183)

**Comment by flozz** [comment]
Ok looks good. I will try to release the fix soon. Thank you for your help! :)

**Comment by flozz** [comment]
**Fix released in v1.2.3.**

The changelog and the Windows builds are available here:

* https://github.com/flozz/yoga-image-optimizer/releases/tag/v1.2.3

The Flatpak package will be available later today :)
